kubernetes-ingress/tls-with-cert-manager.md

```diff
@@ -0,0 +1,121 @@
+self-signed CA using `minica`. In this example, you will be using
+[cert-manager](https://cert-manager.io/), a Kubernetes-native operator that
+issues and renews TLS certificates for you.
+
+## Create TLS certificate and private key
+
+Let us install cert-manager:
+
+```sh
+helm repo add jetstack https://charts.jetstack.io
+helm install cert-manager jetstack/cert-manager --version v1.7.1 --namespace cert-manager --set installCRDs=true --create-namespace
+```
+
+Now, create a CA Issuer:
+
+```sh
+kubectl apply -f https://raw.githubusercontent.com/cilium/cilium-service-mesh-beta/main/kubernetes-ingress/ca-issuer.yaml
+```
+
+## Deploy the ingress
+
+If you previously ran the HTTP and/or gRPC ingress demos, delete the ingresses:
+
+```sh
+kubectl delete ingress basic-ingress grpc-ingress
+```
+
+The ingress configuration for this demo provides the same routing as those demos
+but with the addition of TLS termination. Deploy this ingress:
+
+```sh
+kubectl apply -f https://raw.githubusercontent.com/cilium/cilium-service-mesh-beta/main/kubernetes-ingress/tls-ingress.yaml
+```
+
+This creates a LoadBalancer service, which after around 30 seconds or so should
+be populated with an external IP address.
+
+```console
+$ kubectl get ingress
+NAME          CLASS    HOSTS                                            ADDRESS        PORTS     AGE
+tls-ingress   cilium   hipstershop.cilium.rocks,bookinfo.cilium.rocks   35.195.24.75   80, 443   6m5s
+```
+
+To tell cert-manager that this Ingress needs a certificate, annotate the
+Ingress with the name of the CA issuer we previously created:
+
+```sh
+kubectl annotate ingress tls-ingress cert-manager.io/issuer=ca-issuer
+```
+
+This creates a Certificate object along with a Secret containing the TLS
+certificate.
+
+```console
+$ kubectl get certificate,secret demo-cert
+NAME                                    READY   SECRET      AGE
+certificate.cert-manager.io/demo-cert   True    demo-cert   33m
+
+NAME               TYPE                DATA   AGE
+secret/demo-cert   kubernetes.io/tls   3      33m
+```
+
+## Edit /etc/hosts
+
+In this ingress configuration, the host names `hipstershop.cilium.rocks` and
+`bookinfo.cilium.rocks` are specified in the path routing rules. The client
+needs to specify which host it wants to access. This can be achieved by
+editing your local `/etc/hosts` file. (You will almost certainly need to be
+superuser to edit this file.) Add entries using the IP address
+assigned to the ingress service, so your file looks something like this:
+
+```text
+##
+# Host Database
+#
+# localhost is used to configure the loopback interface
+# when the system is booting.  Do not change this entry.
+##
+127.0.0.1       localhost
+255.255.255.255 broadcasthost
+...
+34.79.55.0 bookinfo.cilium.rocks
+34.79.55.0 hipstershop.cilium.rocks
+```
+
+## Make requests
+
+By specifying the CA's certificate on a curl request, you can say that you trust
+certificates signed by that CA.
+
```

Comment:
```suggestion

This is very similar to the basic TLS demo, except that the CA's secret needs to be retrieved from the Kubernetes secret where it's stored. 
```

Reply:
Done in 06045ec210bbea3246355602364105b6df3e3e0a, thank you!!